### DISCUSSION TITLE
Another font question

### DISCUSSION THREAD

**Discussion by 1stevengrant** [stefanzweifel/sidecar-browsershot#44] [Q&A] [answered]
What's the difference between the package and Spatie's actual Browsershot package in terms of how Puppeteer works under the hood?

As an example, I passed a url to process and Arial font didn't display as well as some alignment issues as well. I actually tested Screeenly and it had the same issues.

I used standalone Browsershot with local Puppeteer and everything worked as expected.

**Comment by stefanzweifel** [comment]
Hm. Could you maybe a share a code snippet on how to reproduce your issue?
3.screeenly.com uses sidecar-browsershot, so you're seeing the same bug. :)

---

In terms of difference between this and the Spatie package:

- BrowsershotLambda: We override the [`callBrowser`](https://github.com/stefanzweifel/sidecar-browsershot/blob/main/src/BrowsershotLambda.php#L15-L41) method, to call the Lambda function instead of the local puppeteer.
- The [BrowsershotFunction](https://github.com/stefanzweifel/sidecar-browsershot/blob/main/src/Functions/BrowsershotFunction.php) is special to the sidecar variant and doesn't exist in the Spatie package. It [updates](https://github.com/stefanzweifel/sidecar-browsershot/blob/main/src/Functions/BrowsershotFunction.php#L47-L51) the call to puppeteer with a AWS Lambda version of puppeteer.
- [browsershot.js](https://github.com/stefanzweifel/sidecar-browsershot/blob/main/resources/lambda/browsershot.js) also doesn't exist in the Spatie package. It stores the options on the local filessystem of AWS and then [executes](https://github.com/stefanzweifel/sidecar-browsershot/blob/main/resources/lambda/browsershot.js#L45-L49) Spatie's `browser.js`.


So basically the only difference is the used version of Puppeteer. https://github.com/stefanzweifel/sidecar-browsershot/blob/main/src/Functions/BrowsershotFunction.php#L47-L51

---

In the last few weeks I also ran into some weird bugs with the Lambda version of Browsershot. The local Chromium version always worked.
I made a PR to the layer repository to update the underlying version of Chromium, but no feedback so far: https://github.com/shelfio/chrome-aws-lambda-layer/pull/54

Currently thinking about deploying my own layers to all AWS region and building an automated process to keep them always up to date.

**Reply by Loots-it** [reply]
I just discovered sidecar-browsershot and the simplicity is just amazing. But I do run into the same issues as @1stevengrant.

**Reply by stefanzweifel** [reply, marked as answer]
For completeness: An issue with more details was created in #97, where we discovered that some fonts – like Arial – are not available by default on AWS Lambda.

In #101 we've added support to bundle custom fonts with the Lambda function. Fonts can be placed in `resources/sidecar-browsershot/fonts`. During deployment, these fonts are bundled with the function and deployed to AWS Lambda.

More on this in the README: https://github.com/stefanzweifel/sidecar-browsershot?tab=readme-ov-file#register-custom-fonts
